flink-core/src/main/java/org/apache/flink/api/java/typeutils/runtime/NullableSerializer.java

```diff
@@ -116,7 +139,7 @@ public T copy(T from, T reuse) {
 	@Override
 	public int getLength() {
 		int len = originalSerializer.getLength();
-		return len < 0 ? len : len + 1;
+		return len == 0 ? 1 : -1;
```

Comment:
This change seems to be a bug.

Reply:
Followup question: how did it pass the tests?

Reply:
The serializer does not have fixed length in common case now (1 or 1 + original length). It is fixed to 1 only of original length is zero. Let's add an option to pad the null value and preserve the fixed length if original serializer has the fixed length.